--- a/net/minecraft/world/level/LevelWriter.java
+++ b/net/minecraft/world/level/LevelWriter.java
@@ -6,9 +_,9 @@
 import net.minecraft.world.level.block.state.BlockState;
 
 public interface LevelWriter {
-    boolean setBlock(BlockPos p_46947_, BlockState p_46948_, int p_46949_, int p_46950_);
+    boolean setBlock(BlockPos p_46947_, BlockState p_46948_, @net.neoforged.neoforge.annotations.BlockFlags int p_46949_, int p_46950_);
 
-    default boolean setBlock(BlockPos p_46944_, BlockState p_46945_, int p_46946_) {
+    default boolean setBlock(BlockPos p_46944_, BlockState p_46945_, @net.neoforged.neoforge.annotations.BlockFlags int p_46946_) {
         return this.setBlock(p_46944_, p_46945_, p_46946_, 512);
     }
 
